server/queue/queue-jms/src/main/java/org/apache/james/queue/jms/JMSMailQueue.java

```diff
@@ -232,7 +232,6 @@ public void enQueue(Mail mail, long delay, TimeUnit unit) throws MailQueueExcept
             enqueuedMailsMetric.increment();
             mailQueueSize.increment();
         } catch (Exception e) {
-            rollback(session);
```

Comment:
what do you think of Spring approach on this subject? https://docs.spring.io/spring/docs/2.0.x/javadoc-api/org/springframework/jms/support/JmsUtils.html#rollbackIfNecessary(javax.jms.Session)
```
	/**
	 * Rollback the Session if not within a JTA transaction.
	 * @param session the JMS Session to rollback
	 * @throws JMSException if committing failed
	 */
	public static void rollbackIfNecessary(Session session) throws JMSException {
		Assert.notNull(session, "Session must not be null");
		try {
			session.rollback();
		}
		catch (javax.jms.TransactionInProgressException ex) {
			// Ignore -> can only happen in case of a JTA transaction.
		}
		catch (javax.jms.IllegalStateException ex) {
			// Ignore -> can only happen in case of a JTA transaction.
		}
	}
```

Reply:
What's the point of a rollback on something not transacted? Does this even make sens?

For me, it just bring additionnal confusion. It can lead to think that **enqueue** can be rollbacked, and might be transacted (which is not).

The only bonus point is if we decide to make it be transacted. Which we are not likely to do for performance.

(So I don't like the Spring version much. Let's avoid the invalid thing in the first place rather than deal with it's consequences.)

Reply:
What does all that mean ? What is a transactional message enqueue ? It just means that we have to call commit later ? What is this performance penalty you are talking about ?
Sorry but I think this is rather light in term of explanation (the ticket is not helpful neither).

Reply:
```
 What is a transactional message enqueue ?
```

I guess you want it when you want to be sure N messages are published (or none of them).

```
It just means that we have to call commit later ? 
```

No because this class was designed to not commit upon publish (cf the session is not transactional)

```
What is this performance penalty you are talking about ?
```

Transactionallity have a cast, right?

```
Sorry but I think this is rather light in term of explanation (the ticket is not helpful neither).
```

I don't see what to explain. I just see a bad API (JMS) used in a bad way (rollback a non transacted session). And I intend to fix it.

Reply:
Here is how the session is created:

```
JMSMailQueue l220

            session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);

```

BTW https://github.com/linagora/james-project/pull/1288 that you approuved relies on the producer session not being transacted.

Reply:
The answers you gave are the explanations that I lacked.

Reply:
why don't you remove the wrapper for rollback then?

Reply:
Because it is used in other places like message consuming where we rely on transaction sessions...